### ISSUE TITLE
Duplicate protoflux groups maintain connections to original copies

### ISSUE THREAD

**Issue by code807** [Yellow-Dog-Man/Resonite-Issues#984]
### Describe the bug?

When duplicating selected groups of ProtoFlux nodes, the duplicates maintain connections to the original ProtoFlux.

### To Reproduce

- Create and select any multi-node ProtoFlux group
- Duplicate the selection

### Expected behavior

The duplicate nodes should have connections between each other rather than connections to their originals.

### Screenshots

![Resonite_xMXMkragu4](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/10662276/efa54b17-6a48-449d-b46d-e9439529f240)
![Resonite_hsW2s7Pb8W](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/10662276/9c8bba1e-347b-4ba8-8d11-c2b954ac795b)
![Resonite_5yNvX3CgtC](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/10662276/1ac41dd3-1b4a-4f1e-85f3-f95d1a6b4e8c)

### Resonite Version Number

Beta 2023.12.21.1127

### What Platforms does this occur on?

Windows

### Log Files

[DESKTOP-1E2V3A8 - 2023.12.21.1127 - 2023-12-23 00_35_53.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13758518/DESKTOP-1E2V3A8.-.2023.12.21.1127.-.2023-12-23.00_35_53.log)

### Reporters

code807

**Comment by Nytra** [comment]
Possibly related https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/907

**Comment by art0007i** [comment]
> Possibly related #907

I don't think that issue is related at all. I'm pretty sure this issue happens because when you grab multiple items and duplicate them each one is duplicated individually, rather than all at once

**Comment by JackTheFoxOtter** [comment]
I would definitely consider this a bug. You're holding a group of objects with internal references. Duplicating them should properly re-assign the references in the same way references withing duplicated hierarchies are reassigned everywhere else in Resonite.

It is also currently impossible to hold a group of ProtoFlux and bring it with you into a different world by focusing there, I assume for the same underlying issue this ticket is describing.
